components/Hero/Hero.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Button } from "../Button/Button";
+import Image from "next/image";
+
+export interface HeroProps extends React.HTMLProps<HTMLElement> {
+  title: string;
+  subTitle: string;
+  image: {
+    src: string;
+    alt: string;
+  };
+}
+
+export const Hero = ({ title, subTitle, image }: HeroProps) => {
+  return (
+    <section className="relative py-10">
+      <div className="flex max-w-screen-xl flex-col lg:ml-auto lg:flex-row lg:items-center">
+        <div className="mb-14 mr-2 lg:w-6/12">
+          <h1 className="max-w-4xl text-blue">{title}</h1>
+          <p className="my-4 mb-9 font-inter text-xl font-medium text-blue">
+            {subTitle}
+          </p>
+          <Button appearance="button">Ver juguetes</Button>
+        </div>
+        <Image
+          src={image.src}
```

Comment:
In the `src` we are gonna be needing the [`urlFor` function](https://github.com/Juguetear/juguetear-web/blob/8cf04f073ec7303a34bc9f0ab65af1d037a902bd/lib/client.ts#L16-L17). If you want, you can [read this](https://www.sanity.io/docs/image-url#usage) to see examples on usage.

Also, here are the types/props of the HeroSeccion
```ts
export interface HeroSection {
  cta: CTA
  description: string
  photo: Photo
  title: string
}
export interface CTA {
  _type: string
  name: string
  url: string
}
export interface Photo { // We can pass this entire object to 'urlFor'
  _type: string
  altText: string
  asset: Asset
}
export interface Asset {
  _ref: string
  _type: string
}
```

Reply:
One question, each field in this objects is required, right?

Reply:
> Adding, check the margin on the left, in bigger screens it makes the content go way to the right when it should be centered

I did `lg:ml-auto` on a container (you mean this @TonyMckes , right?) to do `position: relative` on the image and keep it responsive 
and not apply extreme `right` values to push the image to the right side as in design.

Reply:
> Yes. Though they may change in the future

What fields `_type`, `name`, and`_ref` are for?

> Try with `lg:mx-auto`, and maybe showing the whole picture (on bigger screens, it's not in the designs, but I feel it would make sense. Just play around with it and see how it looks)

Ok, got it!

Reply:
> What fields `_type`, `name`, and`_ref` are for?

- `name` in CTA, is like the 'label' that displays in a link/button.
- `_ref` is a [reference](https://www.sanity.io/docs/reference-type) to another document in Sanity, in this case is an Image.
- `_type` is the [schema type](https://www.sanity.io/docs/schema-types).

If it starts with underscores (_), it means that is created by Sanity.

Here are some examples:
```ts
_type
_createdAt
_key
_id
_rev
_type
_updatedAt
```

Reply:
My component looks like this:

```tsx
import React from "react";
import Image from "next/image";
import { Link } from "components/Link/Link";
import { CTA, Photo } from "types/layout-schema-types";
import { urlFor } from "lib/client";

export interface HeroSection extends React.HTMLProps<HTMLElement> {
  cta: CTA;
  description: string;
  photo: Photo;
  title: string;
}

export const Hero = ({ title, description, photo, cta }: HeroSection) => {
  return (
    <section className="relative py-10">
      <div className="flex max-w-screen-xl flex-col overflow-hidden lg:mx-auto lg:flex-row lg:items-center">
        <div className="mb-14 lg:mb-0 lg:w-5/12">
          <h1 className="mb-1 max-w-4xl text-blue md:mb-5">{title}</h1>
          <p className="mb-8 font-inter text-base font-medium text-blue md:mb-9 md:text-xl">
            {description}
          </p>
          <Link href={cta.url} appearance="button">
            {cta.name}
          </Link>
        </div>
        <Image
          src={urlFor(photo).url()}
          width="750"
          height="478"
          className="relative w-full object-center lg:left-36 lg:w-8/12 lg:rounded-l-[56px] xl:left-40 xl:w-full"
          alt={photo.altText}
        />
      </div>
    </section>
  );
};
```

Is it correct?

Reply:
Yes @alexxuss1996, those are the props

Reply:
@TonyMckes  maybe I can use prop, something like: `imageUrl: urlFor(photo).url()`? That way is much easier to mock data, what do you think? Or there is any way to mock data using `urlFor` function?

Reply:
@alexxuss1996 we can use this for now

```ts
photo: {
  _type: "photo",
  altText: "Hero imagen",
  asset: {
    _ref: "image-2c6ca8b13a29f52d84ed64aca9ef05cc61c1f2b4-681x478-jpg",
    _type: "reference",
  },
},
```

Reply:
> @alexxuss1996 we can use this for now
> 
> ```ts
> photo: {
>   _type: "photo",
>   altText: "Hero imagen",
>   asset: {
>     _ref: "image-2c6ca8b13a29f52d84ed64aca9ef05cc61c1f2b4-681x478-jpg",
>     _type: "reference",
>   },
> },
> ```

Thank you! It works now!